src/platform/sdl2-platform.c

```diff
@@ -252,11 +252,20 @@ static boolean pollBrogueEvent(rogueEvent *returnEvent, boolean textInput) {
     // ~ for (int i=0; i < 100 && SDL_PollEvent(&event); i++) {
     while (SDL_PollEvent(&event)) {
         if (event.type == SDL_QUIT) {
-            rogue.gameHasEnded = true; // causes the game loop to terminate quickly
-            rogue.nextGame = NG_QUIT; // causes the menu to drop out immediately
-            returnEvent->eventType = KEYSTROKE;
-            returnEvent->param1 = ESCAPE_KEY;
-            return true;
+            // the player wants to close the window...
+            if (rogue.recording) {
+                // auto-save
+                flushBufferToFile();
+                if (rogue.gameInProgress && !rogue.quit && !rogue.gameHasEnded) {
+                    saveGameNoPrompt();
+                } else {
+                    char path[BROGUE_FILENAME_MAX];
+                    saveRecordingNoPrompt(path);
```

Comment:
ah, this argument is unused? we should fix that at some point

Reply:
It is only used (passed to `notifyEvent`) by the web platform.